modules/systemd.nix

```diff
@@ -156,6 +164,20 @@ in
           start is considered successful.
         '';
       };
+
+      sessionVariables = mkOption {
+        default = {};
+        type = types.attrs;
```

Comment:
You could change the type here to
```
type = with types; attrsOf (either bool (either int str));
```
which would lead to more predictable results.

I'll see about updating the bash and zsh modules to have the same type for their `sessionVariables` options.